sklearn/cluster/_agglomerative.py

```diff
@@ -891,6 +903,8 @@ def fit(self, X, y=None):
 
         if return_distance:
             self.distances_ = out[-1]
+
+        if self.n_clusters is None:  # distance_threshold was passed
```

Comment:
Maybe more explicitly,

```suggestion
        if self.distance_threshold not is None:
```

They are checked to be mutually exclusive above.

Reply:
I have never seen the construct `not is None`. Did I assume correctly that this is a type for `is not None`?

If yes, fixed in 61e5c5a.

Reply:
`is not None` is correct. `not is None` is probably a mistake :)